### PR TITLE
Fix error (telemetry) in searchview

### LINKED ISSUE #170076: Cannot read properties of undefined (reading 'getValue')
```javascript
TypeError: Cannot read properties of undefined (reading 'getValue')
at Oc.saveState in src/vs/workbench/contrib/search/browser/searchView.ts:1912:53
at <anonymous> in src/vs/workbench/browser/parts/views/viewPaneContainer.ts:745:37
at Array.forEach (<anonymous>)
at B.saveState in src/vs/workbench/browser/parts/views/viewPaneContainer.ts:745:14
at <anonymous> in src/vs/workbench/common/component.ts:27:9
at u.invoke in src/vs/base/common/event.ts:692:17
at s.deliver in src/vs/base/common/event.ts:899:22
at h.fire in src/vs/base/common/event.ts:860:24
at p.flush in src/vs/platform/storage/common/storage.ts:507:25
at <anonymous> in src/vs/platform/storage/common/storage.ts:281:10
```
[Go to Errors Site](https://vscode-errors.azurewebsites.net/card?ch=6ed4d436a9b6ae131732f4e2a723868231f4473a&bH=ba880f0a-68f7-d6a6-80a9-eb471f222060)

### LINKED ISSUE #170723: An extension has contributed more than the allowable number of actions to a comments menu
- VS Code Version:  1.74
- OS Version:  MacOS 12.6

This error is logged to developer console when I have just 3 commands in recently added `comments/commentThread/additionalActions`. 

I've created a small fork of `comment-sample` extension https://github.com/sergei-dyshel/vscode-extension-samples/tree/additional-actions-bug for reproducing the bug.

Steps to Reproduce:

1.  Run the fork.
2.  Create a note. 2 buttons will appear in the bottom: `Placeholder` and `Resolve`.
3. Click `Resolve`. Button disappears and `An extension has contributed more than the allowable number of actions to a comments menu` error is written to console.
4. Expected: Button `Unresolve` should appear but that doesn't happen.

### PR DESCRIPTION
**Pull request by andreamah**
Fixes https://github.com/microsoft/vscode/issues/170076

### CODE REVIEW

File: src/vs/workbench/contrib/search/browser/searchView.ts [hunk 1]
```diff
@@ -1909,10 +1909,10 @@ export class SearchView extends ViewPane {
...
 
 	public override saveState(): void {
-		const patternExcludes = this.inputPatternExcludes.getValue().trim();
-		const patternIncludes = this.inputPatternIncludes.getValue().trim();
-		const onlyOpenEditors = this.inputPatternIncludes.onlySearchInOpenEditors();
-		const useExcludesAndIgnoreFiles = this.inputPatternExcludes.useExcludesAndIgnoreFiles();
+		const patternExcludes = this.inputPatternExcludes ? this.inputPatternExcludes.getValue().trim() : '';
+		const patternIncludes = this.inputPatternIncludes ? this.inputPatternIncludes.getValue().trim() : '';
+		const onlyOpenEditors = this.inputPatternIncludes ? this.inputPatternIncludes.onlySearchInOpenEditors() : false;
+		const useExcludesAndIgnoreFiles = this.inputPatternExcludes ? this.inputPatternExcludes.useExcludesAndIgnoreFiles() : true;
```
Comment (Tyriar) [inline src/vs/workbench/contrib/search/browser/searchView.ts:1915]: This should be the same:

```suggestion
		const patternExcludes = this.inputPatternExcludes?..getValue().trim() ?? '';
		const patternIncludes = this.inputPatternIncludes?.getValue().trim() ?? '';
		const onlyOpenEditors = this.inputPatternIncludes?.onlySearchInOpenEditors() ?? false;
		const useExcludesAndIgnoreFiles = this.inputPatternExcludes?.useExcludesAndIgnoreFiles() ?? true;
```

**Review by Tyriar — CHANGES_REQUESTED** [review]
You might want to remove the `!` from `inputPatternIncludes` and other properties to catch this sort of thing at compile time.

**Review by Tyriar — DISMISSED** [review]

**Comment by andreamah** [comment]
> You might want to remove the `!` from `inputPatternIncludes` and other properties to catch this sort of thing at compile time.

I think that some of the properties like inputPatternIncludes have the `!` because it gets assigned in `renderBody` instead of the constructor.

**Comment by Tyriar** [comment]
@andreamah sure, but because you're telling typescript that you know better problems like this can slip in. `!` on a property is almost always technical debt imo

**Comment by andreamah** [comment]
> @andreamah sure, but because you're telling typescript that you know better problems like this can slip in. `!` on a property is almost always technical debt imo

created #170907 to address this, it might be a big change that I'd rather do in a separate PR

**Review by Tyriar — APPROVED** [review]

**Pull request merged**